dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java

```diff
@@ -81,6 +253,95 @@ public void shouldGetRules() throws DotSecurityException, DotDataException {
+     * When: User with permission try to get all the rules from host
+     * Should: Return all the rules from the host
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldGetRulesFromHostId() throws DotSecurityException, DotDataException {
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role).nextPersisted();
+        final Host host = new SiteDataGen().nextPersisted();
+        final Rule rule1 = new RuleDataGen().host(host).nextPersisted();
+        final Rule rule2 = new RuleDataGen().host(host).nextPersisted();
+
+        final Host anotherHost = new SiteDataGen().nextPersisted();
+        final Rule rule3 = new RuleDataGen().host(anotherHost).nextPersisted();
+
+        this.addPermission(role, host, false);
+        final List<Rule> rules = rulesAPI.getAllRulesByParent(host.getIdentifier(), user, false);
+
+        assertEquals(2, rules.size());
+
+        final List<String> rulesId = rules.stream().map(rule -> rule.getId()).collect(Collectors.toList());
+        assertTrue(rulesId.contains(rule1.getId()));
+        assertTrue(rulesId.contains(rule2.getId()));
+        assertFalse(rulesId.contains(rule3.getId()));
+    }
+
+
+    /**
+     * Method to Test: {@link RulesAPIImpl#getAllRulesByParent(String, User)} )}
+     * When: User with permission try to get all the rules from host
+     * Should: Return all the rules from the host
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldGetRulesFromPageId() throws DotSecurityException, DotDataException {
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role).nextPersisted();
+        final Host host = new SiteDataGen().nextPersisted();
+        this.addPermission(role, host, false);
+
+        final Template template = new TemplateDataGen().nextPersisted();
+        final Folder folder = new FolderDataGen().nextPersisted();
+        final HTMLPageAsset htmlPageAsset = new HTMLPageDataGen(host, template)
+                .folder(folder)
+                .nextPersisted();
+
+        final Rule rule1 = new RuleDataGen().page(htmlPageAsset).nextPersisted();
+        final Rule rule2 = new RuleDataGen().page(htmlPageAsset).nextPersisted();
+        new RuleDataGen().host(host).nextPersisted();
+
+        final HTMLPageAsset anotherHtmlPageAsset = new HTMLPageDataGen(host, template).nextPersisted();
+        final Rule rule3 = new RuleDataGen().page(anotherHtmlPageAsset).nextPersisted();
+
+        addPermissionToReadRulesFolder(role, folder);
+
+        final List<Rule> rules = rulesAPI.getAllRulesByParent(htmlPageAsset.getIdentifier(), user, false);
+
+        assertEquals(2, rules.size());
+
+        final List<String> rulesId = rules.stream().map(rule -> rule.getId()).collect(Collectors.toList());
```

Comment:
why are you comparing against 3 rules if the assert below verifies that there are just 2 rules in the collection?